applications/IgaApplication/python_scripts/nitsche_stabilization_process.py

```diff
@@ -0,0 +1,76 @@
+
+    def __init__(self, model, params):
+        KratosMultiphysics.Process.__init__(self)
+
+        ## Settings string in json format
+        default_parameters = KratosMultiphysics.Parameters("""{
+            "model_part_condition_name" : "",
+            "eigen_system_settings" : {
+                    "solver_type"           : "feast",
+                    "echo_level"            : 0,
+                    "tolerance"             : 1e-10,
+                    "symmetric"             : true,
+                    "e_min"                 : 0.0,
+                    "e_max"                 : 1.0e20,
+                    "number_of_eigenvalues" : 1,
+                    "subspace_size"         : 1
+            },
+            "number_of_couplings" : 1
+        }""")
+
+        ## Overwrite the default settings with user-provided parameters
+        self.params = params
+        self.params.RecursivelyValidateAndAssignDefaults(default_parameters)
+
+        self.model = model
+        self.model_part_condition = self.model[self.params["model_part_condition_name"].GetString()]
+
+        # Call the Nitsche stabilization model Part process
+        KratosMultiphysics.Process.__init__(self)
+        self.process = IGA.NitscheStabilizationModelPartProcess(self.model_part_condition)
+        self.process.ExecuteInitialize()
+
+        self.model_part = self.model.GetModelPart("IgaModelPart").GetSubModelPart("Nitsche_Stabilization_Coupling_" + self.params["model_part_condition_name"].GetString()[-1])
+
+        # Define the eigenvalue size for FEAST solver
+        eigenvalue_nitsche_stabilization_size = self.model_part.ProcessInfo.GetValue(IGA.EIGENVALUE_NITSCHE_STABILIZATION_SIZE)
+        self.params["eigen_system_settings"]["subspace_size"].SetInt(eigenvalue_nitsche_stabilization_size)
+        self.params["eigen_system_settings"]["number_of_eigenvalues"].SetInt(eigenvalue_nitsche_stabilization_size)
+
+    def ExecuteInitializeSolutionStep(self):
+        # Get the model parts which divide the problem
+        current_process_info = self.model_part.ProcessInfo
+
+        # Compute the eigen values
+        eigen_linear_solver = eigen_solver_factory.ConstructSolver(self.params["eigen_system_settings"])
+        builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(eigen_linear_solver)
+        eigen_scheme = IGA.EigensolverNitscheStabilizationScheme()
+        eigen_solver = IGA.EigensolverNitscheStabilizationStrategy(self.model_part, eigen_scheme, builder_and_solver)
+        eigen_solver.Solve()
+
+        # Compute the Nitsche stabilization factor
+        eigenvalue_nitsche_stabilization_vector = current_process_info.GetValue(IGA.EIGENVALUE_NITSCHE_STABILIZATION_VECTOR)
+        nitsche_stabilization_factor= eigenvalue_nitsche_stabilization_vector[eigenvalue_nitsche_stabilization_vector.Size()-1]*4*self.params["number_of_couplings"].GetInt()
```

Comment:
we should check if the stabilization factor is really linearly dependent on the number of couplings. That sound a bit weird to me...

Reply:
Noted.